owncloudApp/src/androidTest/java/com/owncloud/android/settings/camerauploads/OCSettingsCameraUploadsTest.kt

```diff
@@ -0,0 +1,255 @@
+    private val CAMERA_VIDEO_UPLOADS_WIFI = "camera_video_uploads_on_wifi"
+    private val CAMERA_SOURCE_PATH = "camera_uploads_source_path"
+
+    private lateinit var mPrefCameraPictureUploads: CheckBoxPreference
+    private lateinit var mPrefCameraVideoUploads: CheckBoxPreference
+
+    @Before
+    fun setUp() {
+        //Set preferences as disabled with defaults
+        var preferencesEditor = PreferenceManager.getDefaultSharedPreferences(context).edit();
+        preferencesEditor.putBoolean(CAMERA_PICTURE_UPLOADS, false);
+        preferencesEditor.putBoolean(CAMERA_VIDEO_UPLOADS, false);
+        preferencesEditor.putBoolean(CAMERA_PICTURE_UPLOADS_WIFI, false);
+        preferencesEditor.putBoolean(CAMERA_VIDEO_UPLOADS_WIFI, false);
+        preferencesEditor.commit();
+
+        //To set the initial UI status
+        mPrefCameraPictureUploads = activityRule.activity.findPreference(CAMERA_PICTURE_UPLOADS) as CheckBoxPreference
+        mPrefCameraVideoUploads = activityRule.activity.findPreference(CAMERA_VIDEO_UPLOADS) as CheckBoxPreference
+
+        activityRule.activity.runOnUiThread(Runnable {
+            mPrefCameraPictureUploads.setChecked(false)
+            mPrefCameraVideoUploads.setChecked(false)
+        })
+
+    }
+
+    @Test
+    fun checkTitle() {
+        //Asserts
+        onView(withText(R.string.actionbar_settings)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun pictureUploadsView() {
+        //Asserts
+        onView(withText(R.string.prefs_camera_picture_upload)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_picture_upload_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_picture_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_picture_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun videoUploadsView() {
+        //Asserts
+        onView(withText(R.string.prefs_camera_video_upload)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_video_upload_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_video_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_video_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun optionsCameraFolderBehaviour() {
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .check(doesNotExist())
+        onView(withText(R.string.prefs_camera_upload_behaviour_title)).check(doesNotExist())
+        onView(withText(R.string.pref_behaviour_entries_keep_file)).check(doesNotExist())
+    }
+
+    @Test
+    fun enablePictureUploads() {
+        onView(withText(R.string.prefs_camera_picture_upload)).perform(click());
```

Comment:
You could call here enablePictureSubOptions() and delete these two lines

Reply:
I thought about doing it. The reason i did this is because the methods `enablePictureSubOptions()` and `enableVideoSubOptions()` are intended to set an initial status in the tests. But, in this test (and another ones you have mentioned in your comments), both actions are part of the test itself. You save code lines by calling the methods, but the steps in test are less readable.